src/main/java/com/example/fooddelivery/facade/impl/DefaultOrderFacade.java

```diff
@@ -0,0 +1,94 @@
+package com.example.fooddelivery.facade.impl;
+
+import com.example.fooddelivery.dto.OrderDetailsDto;
+import com.example.fooddelivery.dto.ProductSessionDto;
+import com.example.fooddelivery.enums.DeliveryStatus;
+import com.example.fooddelivery.exceptions.EntityNotFoundException;
+import com.example.fooddelivery.facade.OrderFacade;
+import com.example.fooddelivery.forms.AddressForm;
+import com.example.fooddelivery.model.*;
+import com.example.fooddelivery.repository.OrderRepository;
+import com.example.fooddelivery.repository.ProductRepository;
+import com.example.fooddelivery.repository.UserRepository;
+import com.example.fooddelivery.service.AddressService;
+import com.example.fooddelivery.service.OrderService;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import javax.persistence.EntityExistsException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+
+public class DefaultOrderFacade implements OrderFacade {
+
+    private static final String ORDER_NOT_FOUND = "Order not found";
+
+    @Autowired
+    private OrderService orderService;
+
+    @Autowired
+    private OrderRepository orderRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ProductRepository productRepository;
```

Comment:
Can you move repository inside service?

Reply:
The same problem with facade/service. Let's find some useful links.